src/main/java/org/jenkinsci/plugins/parallel_test_executor/ParallelTestExecutor.java

```diff
@@ -312,4 +341,11 @@ public String getDisplayName() {
             return "Parallel test job execution";
         }
     }
+
+    static Predicate<FlowNode> stageNamePredicate(@Nonnull String stageName) {
+        // Arrays.asList in to prevent compiler warning due to unchecked generics/varargs whackiness.
+        return Predicates.and(Arrays.asList(new NodeStepTypePredicate("stage"),
+                FlowScanningUtils.hasActionPredicate(LabelAction.class),
+                new NodeDisplayNamePredicate(stageName)));
```

Comment:
Why both? If it is a `stage` node we know for sure it has a `LabelAction` right?

Conversely, can we not simply look for any block with `LabelAction`, and just treat `stage` as a typical example?

Reply:
Good question. Let me think on that a bit.

Reply:
Ah, ok, so we need both checks because there are two `stage` start nodes, and we only want the one with the `LabelAction`. And I'd rather be specific to `stage`, at least for now.